lib/rust/lexer/definition/src/lexer.rs

```diff
@@ -293,47 +519,70 @@ impl EnsoLexer {
 
     /// Create a blank identifier from the current match.
     fn on_blank<R:LazyReader>(&mut self, _reader:&mut R) {
+        trace!(self.logger,"Identifier::on_blank");
         let token        = Token::Blank(self.offset.consume());
         let suffix_check = self.ident_suffix_check;
         self.discard_current();
         self.append_token(token);
         self.push_state(suffix_check);
     }
 
+    /// Create an annotation from the current match.
+    fn on_annotation<R:LazyReader>(&mut self, _reader:&mut R) {
+        trace!(self.logger,"Identifier::on_annotation");
+        let current      = self.consume_current();
+        let offset       = self.offset.consume();
+        let token        = Token::Annotation(&current[1..],offset);
+        let suffix_check = self.ident_suffix_check;
+        self.append_token(token);
+        self.push_state(suffix_check);
+    }
+
     /// Tokenize an unexpected error suffix.
     fn ident_on_error_suffix<R:LazyReader>(&mut self, _reader:&mut R) {
+        trace!(self.logger,"Identifier::ident_on_error_suffix");
         let token = Token::InvalidSuffix(self.consume_current(),self.offset.consume());
         self.append_token(token);
         self.pop_state();
     }
 
     /// Submit a non-error identifier.
     fn ident_on_no_error_suffix<R:LazyReader>(&mut self, _reader:&mut R) {
+        trace!(self.logger,"Identifier::ident_on_no_error_suffix");
         self.pop_state();
     }
 
+    fn ident_on_eof_suffix<R:LazyReader>(&mut self, _reader:&mut R) {
+        trace!(self.logger,"Identifier::ident_on_eof_suffix");
+        unimplemented!();
+    }
+
     /// The set of rules for lexing Enso identifiers.
     fn add_identifier_rules(lexer:&mut EnsoLexer) {
-        let body_char      = (EnsoLexer::lower_ascii_letter() | EnsoLexer::ascii_digit()).many();
-        let underscore     = c!('_');
-        let ticks          = c!('\'').many();
-        let init_var_seg   = EnsoLexer::lower_ascii_letter() >> &body_char;
-        let var_seg        = (EnsoLexer::lower_ascii_letter() | EnsoLexer::ascii_digit()) >> &body_char;
-        let init_ref_seg   = EnsoLexer::upper_ascii_letter() >> &body_char;
-        let ref_seg        = (EnsoLexer::upper_ascii_letter() | EnsoLexer::ascii_digit()) >> &body_char;
+        let body_char    = (EnsoLexer::lower_ascii_letter() | EnsoLexer::ascii_digit()).many();
+        let underscore   = c!('_');
+        let ticks        = c!('\'').many();
+        let init_var_seg = EnsoLexer::lower_ascii_letter() >> &body_char;
+        let var_seg      =
+            (EnsoLexer::lower_ascii_letter() | EnsoLexer::ascii_digit()) >> &body_char;
+        let init_ref_seg = EnsoLexer::upper_ascii_letter() >> &body_char;
+        let ref_seg      =
+            (EnsoLexer::upper_ascii_letter() | EnsoLexer::ascii_digit()) >> &body_char;
```

Comment:
make more variables here and just align `=` - this will be so easier to read. Like, define `upper_ascii_alphanum` etc